projects/gateway/pkg/syncer/setup_syncer.go

```
@@ -191,19 +212,62 @@ func RunGateway(opts Opts) error {
 	if err != nil {
 		return err
 	}
-	go errutils.AggregateErrs(opts.WatchOpts.Ctx, writeErrs, eventLoopErrs, "event_loop")
+	go errutils.AggregateErrs(ctx, writeErrs, eventLoopErrs, "event_loop")
 
-	logger := contextutils.LoggerFrom(opts.WatchOpts.Ctx)
+	logger := contextutils.LoggerFrom(ctx)
 
 	go func() {
 		for {
 			select {
 			case err := <-writeErrs:
 				logger.Errorf("error: %v", err)
-			case <-opts.WatchOpts.Ctx.Done():
+			case <-ctx.Done():
 				return
 			}
 		}
 	}()
+
+	validationServerErr := make(chan error, 1)
+	if opts.Validation != nil {
+		validationWebhook, err := k8sadmisssion.NewGatewayValidatingWebhook(
+			ctx,
+			validationSyncer,
+			opts.WatchNamespaces,
+			opts.Validation.ValidatingWebhookPort,
+			opts.Validation.ValidatingWebhookCertPath,
+			opts.Validation.ValidatingWebhookKeyPath,
+		)
+		if err != nil {
+			return errors.Wrapf(err, "creating validating webhook")
+		}
+
+		go func() {
+			// close out validation server when context is cancelled
+			<-ctx.Done()
+			validationWebhook.Close()
+		}()
+		go func() {
+			contextutils.LoggerFrom(ctx).Infow("starting gateway validation server",
+				zap.Int("port", opts.Validation.ValidatingWebhookPort),
+				zap.String("cert", opts.Validation.ValidatingWebhookCertPath),
+				zap.String("key", opts.Validation.ValidatingWebhookKeyPath),
+			)
+			if err := validationWebhook.ListenAndServeTLS("", ""); err != nil && err != http.ErrServerClosed {
+				select {
+				case validationServerErr <- err:
+				default:
+					logger.DPanicw("failed to start validation webhook server", zap.Error(err))
+				}
+			}
+		}()
+	}
+
+	// give the validation server 100ms to start
+	select {
```

Instead of blocking on this can it live in a go routine.
Errors from this server should be logged, I'm not sure this should fail the gateway po

---

the reason for this block is exactly to indicate that gateway setup failed.

since validation is enabled/disabled through the settings, i think we want to fail if the validation server fails to start. it more easily avoids cases where validation silently failed but we didn't catch it